workflow/executor.py

```diff
@@ -410,7 +415,12 @@ def run(self):
                               q.update_step_state(step.idt, WorkflowStepState.IN_PROGRESS.name), readonly=False,
                               ignore_problems=True)
             try:
-                step.execute(self.logger, self.config.env, rai_config)
+                if step.engine_size:
```

Comment:
so if I specify `xs` as the default and one of my steps happens to have `xs` as well, line 421 would throw? I'd expect it to not throw anything and just proceed with the default engine.

Reply:
might be also good to "check" what is needed for the next step, so if it happens to run on the same size as just provisioned we don't delete & reprovision a new one